crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -461,6 +494,66 @@ module sui::sui_system {
         }
     }
 
+    /// Extract required Balance from vector of Coin<SUI>, transfer the remainder back to sender.
+    fun extract_coin_balance(coins: vector<Coin<SUI>>, amount: option::Option<u64>, ctx: &mut TxContext): Balance<SUI> {
+        let total_balance = coin::into_balance(vector::pop_back(&mut coins));
```

Comment:
Maybe we can reuse [join_vec](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/sources/pay.move#L68) here?

Reply:
Done.

One question, why is join_vec an entry function?

Reply:
Only entry functions can be called directly in a transaction. I guess we wanted to support calling join_vec in a transction.